app/workers/publishing_api_html_attachments_worker.rb

```diff
@@ -131,8 +131,14 @@ def deleted_html_attachments
     edition.attachables.flat_map(&:deleted_html_attachments)
   end
 
+  def deleted_html_attachment_ids
+    deleted_html_attachments.map(&:content_id).to_set
+  end
+
   def do_publish(update_type)
-    content_ids_to_remove.each do |content_id|
+    redirect_ids = content_ids_to_remove + deleted_html_attachment_ids
```

Comment:
Given that `content_ids_to_remove` is already combining together `content_ids` to work out what things need redirecting, it seems like it might be clearer to work in the `deleted_html_attachment_ids` at that point?